File: src/main/java/bandit/ProjectService.java
package bandit;

import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.components.State;
import com.intellij.openapi.diagnostic.Logger;
import com.intellij.openapi.project.Project;
import com.intellij.util.xmlb.XmlSerializerUtil;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@State(name="BanditService")
public class ProjectService implements PersistentStateComponent<ProjectService> {
    private static final Logger LOGGER = Logger.getInstance(ProjectService.class.getPackage().getName());

    public String executable = "";
    public String config = "";

    public ProjectService() {
        LOGGER.debug("Bandit ProjectService instantiated.");
    }

    static ProjectService getInstance(Project project) {
        return ServiceManager.getService(project, ProjectService.class);
    }

    @Nullable
    @Override
    public ProjectService getState() {
        return this;
    }

    @Override
    public void loadState(@NotNull ProjectService state) {
        XmlSerializerUtil.copyBean(state, this);
    }
}

File: src/main/java/security/ProjectService.kt
package security

import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.ServiceManager
import com.intellij.openapi.components.State
import com.intellij.openapi.diagnostic.Logger
import com.intellij.openapi.project.Project
import com.intellij.util.xmlb.XmlSerializerUtil

@State(name = "BanditService")
class ProjectService : PersistentStateComponent<ProjectService?> {
    override fun getState(): ProjectService? {
        return this
    }

    override fun loadState(state: ProjectService) {
        XmlSerializerUtil.copyBean(state, this)
    }

    companion object {
        private val LOGGER = Logger.getInstance(ProjectService::class.java.getPackage().name)
        fun getInstance(project: Project?): ProjectService {
            return ServiceManager.getService(project!!, ProjectService::class.java)
        }
    }

    init {
        LOGGER.debug("Python Security ProjectService instantiated.")
    }
}